src/libnnstreamer-edge/nnstreamer-edge-mqtt.c

```diff
@@ -225,22 +316,25 @@ int
 nns_edge_mqtt_close (nns_edge_h edge_h)
 {
   nns_edge_handle_s *eh;
+  nns_edge_broker_s *bh;
   MQTTAsync handle;
   MQTTAsync_disconnectOptions options = MQTTAsync_disconnectOptions_initializer;
+  char *msg;
 
   eh = (nns_edge_handle_s *) edge_h;
 
-  if (!NNS_EDGE_MAGIC_IS_VALID (eh)) {
+  if (!NNS_EDGE_MAGIC_IS_VALID (eh) || !eh->broker_h) {
     nns_edge_loge ("Invalid param, given edge handle is invalid.");
     return NNS_EDGE_ERROR_INVALID_PARAMETER;
   }
 
-  handle = eh->mqtt_handle;
+  bh = (nns_edge_broker_s *) eh->broker_h;
 
-  if (!handle) {
+  if (!bh->mqtt_h) {
     nns_edge_loge ("Invalid state, MQTT connection was not completed.");
-    return NNS_EDGE_ERROR_IO;
+    return NNS_EDGE_ERROR_INVALID_PARAMETER;
   }
```

Comment:
Question) Error code is changed from `NNS_EDGE_ERROR_IO` to `NNS_EDGE_ERROR_INVALID_PARAMETER`.
Is this your intent?

Reply:
Verified.